### PR TITLE
Improve colocate join choose

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1.  Add network start cost
2. Update predicate output rows least one rows

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/cost/CostModel.java|5|5|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java [hunk 1]
```diff
@@ -292,16 +292,15 @@ public CostEstimate visitPhysicalDistribution(PhysicalDistributionOperator node,
...
+                    result = CostEstimate.of(statistics.getOutputSize(outputColumns) *
+                                    Catalog.getCurrentSystemInfo().getBackendIds(true).size(),
+                            statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum,
+                            Math.max(statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum, 1));
                     break;
                 case SHUFFLE:
                 case GATHER:
                     result = CostEstimate.of(statistics.getOutputSize(outputColumns), 0,
-                            statistics.getOutputSize(outputColumns));
+                            Math.max(statistics.getOutputSize(outputColumns), 1));
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java:303]: This method already guarantee the minimum of outputSize is 1?

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java:303]: No, really output maybe zero, I just want to the network has some start cost at least

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 10 / 10 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/cost/CostModel.java|5|5|100.00%|

**Review by Youngwb — APPROVED** [review]

**Pull request merged**